### ISSUE TITLE
Load remote audio faster iOS using automaticallyWaitsToMinimizeStalling

### ISSUE THREAD

**Issue by YaarPatandarAA** [ryanheise/just_audio#43]
[automaticallyWaitsToMinimizeStalling](https://developer.apple.com/documentation/avfoundation/avplayer/1643482-automaticallywaitstominimizestal?language=objc)

A Boolean value that indicates whether the player should automatically delay playback in order to minimize stalling.

**Comment by YaarPatandarAA** [comment]
Swift Example:
```
// Setup player
player = AVPlayer.init(playerItem: playerItem)
player.automaticallyWaitsToMinimizeStalling = false
```

Obj-C Example:
```
AVPlayer *player = [[AVPlayer alloc] initWithPlayerItem:playerItem];
player.automaticallyWaitsToMinimizeStalling = false;
```

Should be where the player item is init

**Comment by ryanheise** [comment]
I would consider this as an iOS-specific option, with the current behaviour being the default.

**Comment by YaarPatandarAA** [comment]
Yeah even if it is added as a option that would be great. It helps a lot with loading remote audio files quicker on iOS. 

I have used this to reduce some urls which take 10+ seconds to start playing down to 3 seconds to start playing the audio file.

**Comment by Camerash** [comment]
I have made a PR that implements this feature. Can you try it out on my branch? @YaarPatandarAA

**Comment by Camerash** [comment]
Notice that you need to call this method AFTER setting the url, or else the `AVPlayer` instance in iOS native will be null before you set this option, rendering it useless

**Comment by ryanheise** [comment]
@Camerash Although I haven't used this method yet, I believe it should still be useful in the spirit of @YaarPatandarAA 's example code if you call it after calling `setUrl` but without awaiting the result of `setUrl`.

Another way to make the code more robust could be to allow this method to be called before `setUrl`. On the iOS side, define a new boolean field to store the set value of this flag, and conditionally, if `setUrl` has already completed (by inspecting the state), set the `automaticallyWaitsToMinimizeStalling` property on the player. Then in the iOS implementation of `setUrl`, if the flag has been set, then set the `automaticallyWaitsToMinimizeStalling` property on the player. That way, `automaticallyWaitsToMinimizeStalling` will either happen immediately if called after `setUrl` has completed, or it will just record it effectively as pending and will be processed on the next `setUrl`.

**Comment by Camerash** [comment]
@ryanheise I have implemented your suggestion to make the feature more robust 🙏

**Comment by YaarPatandarAA** [comment]
@Camerash works like a charm. Grabbed it from you branch and tried it out. 

@ryanheise can we please get this implemented. I would like to use this in my app.

**Comment by ryanheise** [comment]
Thanks @Camerash , I have now merged this PR and it is published in version 0.1.1. @YaarPatandarAA please feel welcome to reopen if the released version does not work.
